TESTS/mbed_hal/ticker_api_tests/main.cpp

```diff
@@ -0,0 +1,520 @@
+}
+
+/* Test that ticker can be rescheduled repeatedly before the handler has been called.
+ *
+ * Given ticker is available, initialised.
+ * When ticker interrupt is set and then rescheduled (interrupt time is modified).
+ * Then ticker interrupt is triggered according the rescheduled time.
+ */
+void ticker_repeat_reschedule_test(void)
+{
+    /* Clear ISR call counter. */
+    intFlag = 0;
+
+    /* Init the ticker. */
+    intf->init();
+
+    /* Get current tick count. */
+    const uint32_t tick_count = intf->read();
+
+    /* Set interrupt. Interrupt should be fired when tick count is equal to:
+     * tick_count + TICKER_INT_VAL. */
+    intf->set_interrupt(tick_count + TICKER_INT_VAL);
+
+    /* Reschedule interrupt - it should not be fired yet.
+     * Call intf->set_interrupt again. */
+    intf->set_interrupt(tick_count + (2 * TICKER_INT_VAL));
+
+    /* Wait until ticker count reach value: tick_count + 2*TICKER_INT_VAL - TICKER_DELTA.
+     * Interrupt should not be fired. */
+    while (intf->read() < (tick_count + 2 * TICKER_INT_VAL - TICKER_DELTA)) {
+        /* Indicate failure if interrupt has fired earlier. */
+        if (intFlag != 0) {
+            TEST_ASSERT(0);
+        }
+    }
+
+    /* Wait until ticker count reach value: tick_count + 2*TICKER_INT_VAL + TICKER_DELTA.
+     * Interrupt should be fired after this time. */
+    while (intf->read() < (tick_count + 2 * TICKER_INT_VAL + TICKER_DELTA)) {
+        /* Just wait. */
+    }
+
+    /* Check if interrupt has been fired. */
+    TEST_ASSERT_EQUAL(1, intFlag);
+}
+
+/* Test that ticker_fire_interrupt causes and interrupt to get fired immediately.
+ *
+ * Given ticker is available.
+ * When ticker_fire_interrupt is called.
+ * Then ticker interrupt is triggered.
+ */
+void ticker_fire_now_test(void)
+{
+    /* Clear ISR call counter. */
+    intFlag = 0;
+
+    /* Init the ticker. */
+    intf->init();
+
+    /* Fire the interrupt. */
+    intf->fire_interrupt();
+
+    /* Check if interrupt has been fired. */
+    TEST_ASSERT_EQUAL(1, intFlag);
+}
+
+/* Test that the ticker correctly handles overflow.
+ *
+ * Note that for high frequency timers we will only prove that ticker counter rollovers and
+ * continue counting (it is not possible to prove in deterministic way that after rollover next value is 0).
+ *
+ * Given ticker is available.
+ * When ticker has overflows.
+ * Then ticker continues counting from the beginning and interrupt scheduling works.
+ */
+void ticker_overflow_test(void)
+{
+    /* Get ticker info. */
+    const ticker_info_t* p_ticker_info = intf->get_info();
+
+    /* We need to check how long it will take to overflow.
+     * We will perform this test only if this time is no longer than 30 sec.
+     */
+    const uint32_t max_count = (1 << p_ticker_info->bits) - 1;
+    const uint32_t required_time_sec = (max_count / p_ticker_info->frequency);
+
+    if (required_time_sec > 30 && !FORCE_OVERFLOW_TEST) {
+        TEST_ASSERT(true);
```

Comment:
If the specified conditions are fulfilled, then this test case should be omitted. TEST_ASSERT(true) has been called to show that it is assumed that this test case has passed. 
Changed to TEST_ASSERT(true) to TEST_PASS().

Reply:
Since the test case is omitted (not passed) I think `TEST_IGNORE` is more appropriate than `TEST_PASS`, i.e. `TEST_IGNORE_MESSAGE("Ignored -- would take more than 30 s to overflow")`.